common/src/main/scala/org/apache/celeborn/common/util/Utils.scala

```diff
@@ -415,8 +415,33 @@ object Utils extends Logging {
     customHostname = Some(hostname)
   }
 
-  def localHostName(conf: CelebornConf): String = customHostname.getOrElse {
-    if (conf.bindPreferIP) {
+  def localHostName(conf: CelebornConf): String = {
+    getHostName(conf.bindPreferIP)
+  }
+
+  def localHostNameForAdvertiseAddress(conf: CelebornConf, env: String): String = {
+    if (env.equals(RpcNameConstants.MASTER_SYS) || env.equals(
+        RpcNameConstants.MASTER_INTERNAL_SYS)) {
+      getAdvertiseAddressForMaster(conf)
+    } else {
+      getAdvertiseAddressForWorker(conf)
+    }
+  }
+
+  private def getAdvertiseAddressForMaster(conf: CelebornConf) = {
+    if (conf.masterHost != "<localhost>") {
+      conf.masterHost
```

Comment:
It seems that conf.masterHost is never equal to "<localhost>". Why not directly call getHostName(conf.advertisePreferIP) instead?

Reply:
this is to cover this unit test case: https://github.com/apache/celeborn/blob/main/worker/src/test/scala/org/apache/celeborn/service/deploy/MiniClusterFeature.scala#L61

Reply:
the check should maybe be: 
```
    if (conf.masterHost == "localhost") {
      conf.masterHost
    } else {
      getHostName(conf.advertisePreferIP)
    }
```
@RexXiong wdyt?

Reply:
Perhaps you mean that if celeborn.master.host is set, we should prioritize using that. If that’s the case, we can define a new method for this.

```
      def advertiseMasterHost: String = get(MASTER_HOST).replace(<localhost>", Utils.getHostName(this.advertisePreferIP))
```

Reply:
sure i am fine with this

Reply:
updated